File: client/src/app/models/order.ts
export interface Order {
    id: number
    buyerId: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    subTotal: number
    deliveryFee: number
    orderStatus: string
    total: number
  }
  
  export interface ShippingAddress {
    fullName: string
    line1: string
    line2: string
    city: string
    state: string
    zip: string
    country: string
  }
  
  export interface OrderItem {
    productId: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }

File: API/Entities/OrderAggregation/Order.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities.OrderAggregation
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        [Required]
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public List<OrderItem> OrderItems { get; set; }
        public long SubTotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderState { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }

        public long CalcTotal()
        {
            return SubTotal + DeliveryFee;
        }

    }
}